libs/wingc/src/jsify.rs

```diff
@@ -1961,3 +1984,51 @@ fn jsify_type_name(t: &Vec<Symbol>, phase: Phase) -> String {
 		p.join(".")
 	}
 }
+
+fn lookup_span(span: &WingSpan, source_path: &Path) -> String {
+	let source = std::fs::read_to_string(source_path)
```

Comment:
Does it make sense to read this from a file every time we are doing a lookup? Maybe we can just store the source somewhere and propagate down?

How about creating something like `CompilerContext` object that maintains all the sources as well as e.g. `Types`?

(I am less concerned about performance and more about the fact that very soon we will have multiple sources and then it makes sense to manage them in a centralized facility).

Reply:
Updated to use the `Files` utility. I like the compiler context object idea though we have another object also already named `CompilationContext` and I'm not sure if I should be touching it because it's part of a thread_local